Cannot build-frontend with Vaadin 14 beta2 using Java 11

I'm trying to build my own project for npm with Vaadin 14.0.0.beta2 (flow 2.0.0.rc2) using Java 11 (openjdk) and I get the following error from bytebuddy (asm) classes:

`org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.vaadin:vaadin-maven-plugin:14.0.0.beta2:build-frontend (default) on project smp-vaadin: Execution default of goal com.vaadin:vaadin-maven-plugin:14.0.0.beta2:build-frontend failed: This feature requires ASM7
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal com.vaadin:vaadin-maven-plugin:14.0.0.beta2:build-frontend failed: This feature requires ASM7
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.UnsupportedOperationException: This feature requires ASM7
    at net.bytebuddy.jar.asm.ClassVisitor.visitNestMember (ClassVisitor.java:236)
    at net.bytebuddy.jar.asm.ClassReader.accept (ClassReader.java:660)
    at net.bytebuddy.jar.asm.ClassReader.accept (ClassReader.java:400)
    at com.vaadin.flow.server.frontend.FrontendDependencies.visitClass (FrontendDependencies.java:394)
    at com.vaadin.flow.server.frontend.FrontendDependencies.computeEndpoints (FrontendDependencies.java:236)
    at com.vaadin.flow.server.frontend.FrontendDependencies.<init> (FrontendDependencies.java:139)
    at com.vaadin.flow.server.frontend.NodeTasks.<init> (NodeTasks.java:259)
    at com.vaadin.flow.server.frontend.NodeTasks.<init> (NodeTasks.java:38)
    at com.vaadin.flow.server.frontend.NodeTasks$Builder.build (NodeTasks.java:145)
    at com.vaadin.flow.plugin.maven.BuildFrontendMojo.runNodeUpdater (BuildFrontendMojo.java:169)
    at com.vaadin.flow.plugin.maven.BuildFrontendMojo.execute (BuildFrontendMojo.java:126)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)`

The issue seems to be caused by the class com.vaadin.flow.server.frontend.FrontendClassVisitor which is calling the super constructor with Opcodes.ASM6 and at least Opcodes.ASM7 is expected.

I can get the same error by running jetty:run from the Vaadin 14 starter project.

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendAnnotatedClassVisitor.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendAnnotatedClassVisitor.java
@@ -53,7 +53,7 @@ class FrontendAnnotatedClassVisitor extends ClassVisitor {
      *            The annotation class name to visit
      */
     FrontendAnnotatedClassVisitor(ClassFinder finder, String annotationName) {
-        super(Opcodes.ASM6);
+        super(Opcodes.ASM7);
         this.finder = finder;
         this.annotationName = annotationName;
     }

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendClassVisitor.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendClassVisitor.java
@@ -50,7 +50,7 @@ class FrontendClassVisitor extends ClassVisitor {
      */
     static class RepeatedAnnotationVisitor extends AnnotationVisitor {
         public RepeatedAnnotationVisitor() {
-            super(Opcodes.ASM6);
+            super(Opcodes.ASM7);
         }
 
         @Override
@@ -159,12 +159,12 @@ class FrontendClassVisitor extends ClassVisitor {
      *            the end-point object that will be updated during the visit
      */
     FrontendClassVisitor(String className, EndPointData endPoint) { //NOSONAR
-        super(Opcodes.ASM6);
+        super(Opcodes.ASM7);
         this.className = className;
         this.endPoint = endPoint;
 
         // Visitor for each method in the class.
-        methodVisitor = new MethodVisitor(Opcodes.ASM6) {
+        methodVisitor = new MethodVisitor(Opcodes.ASM7) {
             // We are interested in the new instances created inside the method
             @Override
             public void visitTypeInsn(int opcode, String type) {

